src/main/scala/io/iohk/ethereum/consensus/pow/EthashUtils.scala

```diff
@@ -63,12 +63,18 @@ object EthashUtils {
 
   private def epochBeforeEcip1099(blockNumber: Long): Long = blockNumber / EPOCH_LENGTH_BEFORE_ECIP_1099
 
-  def seed(blockNumber: Long): ByteString = {
-    val epoch = epochBeforeEcip1099(blockNumber)
+  def seed(blockNumber: Long, ecip1099ActivationBlock: Long): ByteString = {
+    val epoch: Long = epochBeforeEcip1099(startBlock(blockNumber, ecip1099ActivationBlock))
     (BigInt(0) until epoch)
       .foldLeft(ByteString(Hex.decode("00" * 32))) { case (b, _) => kec256(b) }
   }
 
+  def startBlock(blockNumber: Long, ecip1099ActivationBlock: Long): Long =
```

Comment:
this is quite confusing - why such a convoluted computation?
can `calcEpochLength(blockNumber, ecip1099ActivationBlock)` change 
in this multiplication `calcEpochLength(blockNumber, ecip1099ActivationBlock)) * calcEpochLength(blockNumber, ecip1099ActivationBlock)`?

perhaps
```scala
val epochLength = calcEpochLength(blockNumber, ecip1099ActivationBlock)
((blockNumber / epochLength) * epochLength) + 1
```
or maybe even more explicitly:
```scala
val epochLength = calcEpochLength(blockNumber, ecip1099ActivationBlock)
val epoch: Long = blockNumber / epochLength
(epoch * epochLength + 1) / EPOCH_LENGTH_BEFORE_ECIP_1099
```
to avoid this indirection:
`epochBeforeEcip1099(startBlock(blockNumber, ecip1099ActivationBlock))`
?

Reply:
Deduplicating that `calcEpochLength(...)` call sounds good, but I'm not sure your other proposals are equivalent.

Reply:
I agree, maybe also the name should change to startEpochBlock, or add a comment, so we have more context when we look at it

Reply:
I merged it all into the `seed` function, since it's only used there anyway.